OpenRA.Mods.Common/Activities/FindResources.cs

```diff
@@ -21,6 +20,18 @@ namespace OpenRA.Mods.Common.Activities
 {
 	public class FindResources : Activity
 	{
+		ResourceLayer resLayer;
+		HarvesterInfo harvInfo;
+		ResourceClaimLayer territory;
+		Mobile mobile;
+		Harvester harv;
+		MobileInfo mobileInfo;
+		IPathFinder pathFinder;
+
+		// TODO: This is a hack because this class is sometimes
+		// constructed by reflection and cannot add an Actor variable in
```

Comment:
I would prefer it if we could fix this first in a separate pr.  We can replace the `InitialActivity` hack on `FreeActor` with code in `Harvester` that will queue this activity if it isn't suppressed by an ActorInit or a rules flag (and maybe tie this in to whatever logic makes harvs auto-harvest when built normally?)